hydra-node/exe/hydra-node/Main.hs

```diff
@@ -101,8 +101,12 @@ main = do
                 RunOptions{apiHost, apiPort} = opts
             apiPersistence <- createPersistenceIncremental $ persistenceDir <> "/server-output"
             withAPIServer apiHost apiPort party apiPersistence (contramap APIServer tracer) chain pparams (putEvent . ClientEvent) $ \server -> do
+
               -- Network
-              withNetwork tracer (connectionMessages server) signingKey otherParties host port peers nodeId putNetworkEvent $ \hn -> do
+              msgPersistence <- createPersistenceIncremental $ persistenceDir <> "/network-messages"
+              ackPersistence <- createPersistence $ persistenceDir <> "/acks"
+
+              withNetwork tracer msgPersistence ackPersistence (connectionMessages server) signingKey otherParties host port peers nodeId putNetworkEvent $ \hn -> do
```

Comment:
I think `withNetwork` is only used here, and is the node's network stack which always require persistence. Rather than creating the `Persistence` handles outside, why not pass a directory and let `withNetwork` handle the creation of the files?